app/precompiled.py

```diff
@@ -229,26 +237,24 @@ def overlay_template_areas(src_pdf, page_number=None, overlay=True):
     return png_from_pdf(pdf, page_number)
 
 
-def get_invalid_pages_with_message(src_pdf):
-    message = ""
-    invalid_pages = []
-    invalid_pages = _get_pages_with_invalid_orientation_or_size(src_pdf)
-    if len(invalid_pages) > 0:
-        message = "Your letter is not A4 portrait size on "
-    else:
+def get_invalid_pages_with_errors(src_pdf):
+    passed_validation = False
+    errors = {
+        "content_outside_of_printable_area": [],
+        "document_not_a4_size_portrait_orientation": [],
+    }
+    errors["document_not_a4_size_portrait_orientation"] = _get_pages_with_invalid_orientation_or_size(src_pdf)
+    if not errors["document_not_a4_size_portrait_orientation"]:
         pdf_to_validate = _overlay_printable_areas(src_pdf)
-        invalid_pages = list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
-        if len(invalid_pages) > 0:
-            message = 'Content in this PDF is outside the printable area on '
-    if len(invalid_pages) > 0:
-        message += unescaped_formatted_list(
-            invalid_pages,
-            before_each='',
-            after_each='',
-            prefix='page',
-            prefix_plural='pages'
-        )
-    return (invalid_pages, message)
+        errors["content_outside_of_printable_area"] = list(_get_out_of_bounds_pages(PdfFileReader(pdf_to_validate)))
+        if not errors["content_outside_of_printable_area"]:
+            passed_validation = True
+    return passed_validation, errors
```

Comment:
dont think you really need to keep track of this `passed_validation` variable - you can just check if any of the lists have data in when returning.

 i think this'll work: `return not any(errors.values()), errors`. though that looks a bit magic, unsure how it'll look.